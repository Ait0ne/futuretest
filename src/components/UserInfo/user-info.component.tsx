import React from 'react';
import { IUser } from '../UsersListProvider/users-list-provide.component';

import {BoldText, UserInfoContainer} from './user-info.styles';

interface UserInfoProps {
    user: IUser
}


const UserInfo: React.FC<UserInfoProps> = ({user}) => {
    return (
        <UserInfoContainer>
            <span>Выбран пользователь <BoldText>{user.firstName+' '+user.lastName}</BoldText></span>
            <span>Описание:</span>
            <p>{user.description}</p>
            <span>Адрес проживания: <BoldText>{user.address.streetAddress}</BoldText></span>
            <span>Город: <BoldText>{user.address.city}</BoldText></span>
            <span>Провинция/штат: <BoldText>{user.address.state}</BoldText></span>
            <span>Индекс: <BoldText>{user.address.zip}</BoldText></span>
        </UserInfoContainer>
    )
}

export default UserInfo;