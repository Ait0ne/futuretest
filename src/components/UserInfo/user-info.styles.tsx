import styled from'styled-components';

export const UserInfoContainer = styled.div`
display: flex;
flex-direction: column;
align-items: flex-start;
justify-content: center;
padding: 15px;
>p {
    margin: 5px 0;
}
`

export const BoldText = styled.span`
font-weight: bold;
`