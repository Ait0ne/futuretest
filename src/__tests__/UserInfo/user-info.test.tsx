import React from 'react';
import {shallow} from 'enzyme';

import UserInfo from '../../components/UserInfo/user-info.component';
import {testData} from '../../testData';

describe('UserInfo component', () => {
    const userInfo = shallow(<UserInfo user={{...testData[0], id: '123'}}/>)

    it('renders correctly', () => {
        expect(userInfo).toMatchSnapshot()
    })
})
