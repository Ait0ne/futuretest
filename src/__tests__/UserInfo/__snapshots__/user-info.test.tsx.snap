// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component renders correctly 1`] = `
<styled.div>
  <span>
    Выбран пользователь 
    <styled.span>
      Ning Sherman
    </styled.span>
  </span>
  <span>
    Описание:
  </span>
  <p>
    tempor vestibulum sit turpis fringilla dolor velit amet non scelerisque quis mattis curabitur amet tellus donec ac morbi pulvinar lectus tincidunt dolor dolor sed nullam sed fringilla dolor vitae nec ipsum sollicitudin
  </p>
  <span>
    Адрес проживания: 
    <styled.span>
      9317 Nec Ave
    </styled.span>
  </span>
  <span>
    Город: 
    <styled.span>
      Riverview
    </styled.span>
  </span>
  <span>
    Провинция/штат: 
    <styled.span>
      MT
    </styled.span>
  </span>
  <span>
    Индекс: 
    <styled.span>
      43092
    </styled.span>
  </span>
</styled.div>
`;
